plugins/unicodelib/unicodelib.cpp

```diff
@@ -1416,3 +1459,30 @@ UNICODELIB_API void UNICODELIB_CALL ulUnicodeLocaleGetNthWord(unsigned & tgtLen,
     }
 }
 
+UNICODELIB_API void UNICODELIB_CALL ulUnicodeLocaleExcludeNthWord(unsigned & tgtLen, UChar * & tgt, unsigned textLen, UChar const * text, unsigned n, char const * localename)
+{
+    UErrorCode status = U_ZERO_ERROR;
+    Locale locale(localename);
+    RuleBasedBreakIterator* bi = (RuleBasedBreakIterator*)RuleBasedBreakIterator::createWordInstance(locale, status);
```

Comment:
Efficiency: Note this section in the ICU documentation: http://userguide.icu-project.org/boundaryanalysis
"Reuse

It is slow and wasteful to repeatedly create and destroy a BreakIterator when it is not necessary. For example, do not create a separate BreakIterator for each line in a document that is being word-wrapped. Keep around a single instance of a line BreakIterator and use it whenever a line break iterator is needed.
"
I know you based this on an existing function.  That was written 6 years ago before we had a good code review system in place.  I think it might be best to leave it as it is, but fix it with a separate pull request.